### PR TITLE
Freshly coded new client

### PR DESCRIPTION
**Pull request by programatik29**
- Reordered dependencies and removed their patch version.
- Much more readable and less complex client code.
- Better implemented timeouts.
- Better error messages.
- Use `anyhow` on client code.
- Error logging.
- Uses `tokio-native-tls` instead of `tokio-rustls`.

### CODE REVIEW

**Comment by ChillFish8** [comment]
Ill have to review this tomorrow but any reason for dropping rustls for native tls?

**Comment by programatik29** [comment]
Mainly for convenience on my part. Using `native-tls` results in less code.

**Comment by ChillFish8** [comment]
Bump test

File: src/http/mod.rs [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    Body,
+};
+use std::{collections::HashMap, net::SocketAddr, time::Duration};
+use tokio::{
+    io::{AsyncRead, AsyncWrite},
+    net::TcpStream,
+    task::JoinHandle,
+    time::{error::Elapsed, sleep, timeout_at, Instant},
+};
+use tower::{util::ServiceExt, Service};
```
Comment (ChillFish8) [inline src/http/mod.rs:23]: I think rustfmt needs to be ran on these changes. 

My general style with imports is ordered and separated by whitespace:

- `std`
- `3rd party deps` 
- `crate/super/self` imports

**Review by ChillFish8 — CHANGES_REQUESTED** [review]

**Review by ChillFish8 — APPROVED** [review]

**Pull request merged**